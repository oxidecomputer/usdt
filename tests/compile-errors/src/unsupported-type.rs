#![feature(asm)]
#![cfg_attr(target_os = "macos", feature(asm_sym))]

usdt::dtrace_provider!("../../../tests/compile-errors/providers/unsupported-type.d");

fn main() {
    let bad: u8 = 0;
    unsupported::bad!(|| (bad));
}
