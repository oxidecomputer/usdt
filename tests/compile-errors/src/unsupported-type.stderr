error: proc macro panicked
  --> src/unsupported-type.rs:15:1
   |
15 | usdt::dtrace_provider!("../../../tests/compile-errors/providers/unsupported-type.d");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: message: Could not read D source file "../../../tests/compile-errors/providers/unsupported-type.d" in "$WORKSPACE/target/tests/trybuild/compile-errors"

error[E0433]: failed to resolve: use of undeclared crate or module `unsupported`
  --> src/unsupported-type.rs:19:5
   |
19 |     unsupported::bad!(|| (bad));
   |     ^^^^^^^^^^^ use of undeclared crate or module `unsupported`
